zerver/tests/test_outgoing_webhook_system.py

```diff
@@ -155,20 +155,38 @@ def test_pm_to_outgoing_webhook_bot(self, mock_requests_request: mock.Mock) -> N
         self.assertEqual(last_message.content, "Hidley ho, I'm a webhook responding!")
         self.assertEqual(last_message.sender_id, self.bot_profile.id)
         display_recipient = get_display_recipient(last_message.recipient)
-        # The next two lines error on mypy because the display_recipient is of type Union[str, List[Dict[str, Any]]].
-        # In this case, we know that display_recipient will be of type List[Dict[str, Any]].
-        # Otherwise this test will error, which is wanted behavior anyway.
+        # The next two lines error on mypy because the display_recipient is of type Union[str, List[Dict[
+        # str, Any]]]. In this case, we know that display_recipient will be of type List[Dict[str,
+        # Any]]. Otherwise this test will error, which is wanted behavior anyway.
         self.assert_length(display_recipient, 1)  # type: ignore
         self.assertEqual(display_recipient[0]['email'], self.user_profile.email)   # type: ignore
 
     @mock.patch('requests.request', return_value=ResponseMock(200, {"response_string": "Hidley ho, I'm a webhook responding!"}))
     def test_stream_message_to_outgoing_webhook_bot(self, mock_requests_request: mock.Mock) -> None:
+        # If the bot owner or bot is subscribed to the stream
         self.send_stream_message(self.user_profile, "Denmark",
                                  content="@**{}** foo".format(self.bot_profile.full_name),
-                                 topic_name="bar")
+                                 topic_name="bar", bot_reply=True)
         last_message = self.get_last_message()
         self.assertEqual(last_message.content, "Hidley ho, I'm a webhook responding!")
         self.assertEqual(last_message.sender_id, self.bot_profile.id)
         self.assertEqual(last_message.topic_name(), "bar")
         display_recipient = get_display_recipient(last_message.recipient)
         self.assertEqual(display_recipient, "Denmark")
+
+        # If the bot or bot owner is not subscribed to the stream
```

Comment:
This comment is confusing, because you seem to to subscribe the user to the stream just a few lines later.

Would this test have failed before your fix?  You really want to make sure the test is gonna prevent future regressions.

Reply:
The owner of the bot or the bot is not subscribed to the stream. It is the sender which I am subscribing to the stream, unless the sender is not subscribed, the message will not go through.
Yes, I am sure

Reply:
I think there's a way to make this more clear in the code, either by wording it slightly more clearly or using better var names.

Reply:
maybe just say "We will subscribe Hamlet (sender) to the stream, but not the bot or bot owner."